### PR TITLE
Fcb rename 01

### PR DESCRIPTION
**Pull request by andrewbird**

### CODE REVIEW

**Comment by stsp** [comment]
Thanks for putting things together.
But I am inclined to treat this as a bug reports,
rather than the finished fixes.
The pointer hack can be applied as is, as it
just completes the initial hack that is already
applied. I'll revert them both if/when I find the
better solution.
The 12 (or 13) one was clearly an example.
We should either pass the buffer size separately,
or maybe just reference the `SecPathName` buffer
directly, rather than passing the pointers to it?
As for shadowing, could you please clarify what
exactly does it fix? Shadowing should work properly,
how it happens that the return value is different?

**Comment by stsp** [comment]
I opened #35 and #36 to think more about these problems.

**Comment by andrewbird** [comment]
Okay I'll drop the last two, you can fixup, call them what you like as you wish. I'll fix the commit message in the first later. Incidentally, the shadowing worked too well there, when the inner result went out of scope the value set was obviously lost and the function always returned FCB_SUCCESS, hence the problem reported to freedos.

File: kernel/fcbfns.c [hunk 1]
```diff
@@ -530,7 +530,7 @@ UBYTE FcbRename(xfcb FAR * lpXfcb)
   {
     dmatch Dmatch;
     dmatch FAR * Dmatch_p;
-    COUNT result;
+    COUNT _result;
```
Comment (stsp) [inline kernel/fcbfns.c]: Why not to just remove this line and forget?

**Comment by stsp** [comment]
I applied the simpler fix to the shadowing - just removed it. :)

**Comment by stsp** [comment]
Applied the third one by hands.

**Comment by andrewbird** [comment]
> I applied the simpler fix to the shadowing - just removed it. :)

Simpler but doesn't work, returns failure on success. The types of the two result vars are different, you are now testing an unsigned byte var for < 0.
Just to be sure nothing else changed I backed out your simple fix, reapplied mine and it worked as before.

**Comment by stsp** [comment]
Ah, sorry about that.
Only after debugging it myself, I now see
your point, and why it should be the different
var, not the same one.
I rebased my patch out. Please update your
PR and I'll apply.

**Comment by stsp** [comment]
The comment why another var is needed,
would be good. Also I suggest the name `rc`
for new var or alike, to avoid confusion.

**Comment by stsp** [comment]
Applied, thanks.

**Comment by andrewbird** [comment]
> The comment why another var is needed,
would be good. Also I suggest the name rc
for new var or alike, to avoid confusion.

Thanks for the advice, I redid the patch for FreeDOS and it looks lots better with 'rc'.

**Pull request closed without merging**